crates/guest-rust/rt/src/async_support/future_support.rs

```diff
@@ -84,7 +103,11 @@ impl<T> CancelableWrite<T> {
                 Handle::LocalReady(..) => {
                     entry.insert(Handle::LocalOpen);
                 }
-                Handle::Write => (writer.vtable.cancel_write)(writer.handle),
+                Handle::Write => unsafe {
+                    // TODO: spec-wise this can return `BLOCKED` which seems
```

Comment:
It can only return `BLOCKED` if called with the `async` option enabled.  As long as we don't try to import it using the `[async]` prefix, we should be good.

Reply:
Ah ok, I'll leave this as a follow-up of https://github.com/bytecodealliance/wit-bindgen/issues/1188 to come back and fill in comments here.